core/src/main/java/io/questdb/cairo/map/RecordValueSinkFactory.java

```diff
@@ -84,7 +84,8 @@ public static RecordValueSink getInstance(BytecodeAssembler asm, ColumnTypes col
             asm.aload(1);
             asm.iconst(index);
 
-            switch (columnTypes.getColumnType(index)) {
+            int columnType = columnTypes.getColumnType(index);
+            switch (ColumnType.sizeTag(columnType)) {
```

Comment:
`sizeTag()` is untested, realistically this might be needed by `first()` or `last()` functions but there are no tests

Reply:
It's tested in JoinTest, but the test is in Ignored state. To be enabled later on when map and geohash problem is solved